src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStateOpResolverTest.scala

```diff
@@ -23,15 +25,129 @@ class TaskStateOpResolverTest
   test("an update for a non-existing tasks is mapped to fail") {
     val f = new Fixture
     Given("a taskID without task")
-    val appId = PathId("/app")
-    val taskId = Task.Id.forApp(appId)
-    f.taskTracker.task(taskId) returns Future.successful(None)
+    f.taskTracker.task(f.notExistingTaskId) returns Future.successful(None)
 
     When("resolve is called")
-    val stateChange = f.stateOpResolver.resolve(TaskStateOp.ForceExpunge(taskId)).futureValue
+    val stateChange = f.stateOpResolver.resolve(TaskStateOp.ForceExpunge(f.notExistingTaskId)).futureValue
+
+    Then("getTaskAsync is called")
+    verify(f.taskTracker).task(f.notExistingTaskId)
+
+    And("the future fails")
+    stateChange shouldBe a[TaskStateChange.Failure]
+
+    And("there are no more interactions")
+    f.verifyNoMoreInteractions()
+  }
+
+  test("ForceExpunge fails if task does not exist") {
```

Comment:
What is the difference to the last test?

Also, if we use ForceExpunge as a revert and we detect that the task does not exist. Is that really an error? (Corner-case, ignore if you want for now)